core/src/helpers.ts

```diff
@@ -168,6 +172,41 @@ const buildRestList = <Model>(
   });
 };
 
+type TCreateSpecializedTransformersParams<TModel> = {
+  type: 'rest' | 'graphql';
+  buildFields: (keyof TModel)[];
+};
+const createSpecializedTransformers = <TModel>({
+  type,
+  buildFields,
+}: TCreateSpecializedTransformersParams<TModel>) => {
+  return {
+    [type]: Transformer<TModel, TModel>(type, {
+      buildFields: buildFields,
+    }),
+  };
+};
+
+type TCreateSpecializedBuilderParams<TModel> = {
+  generator: TGeneratorResult<TModel>;
+  type: 'rest' | 'graphql';
+  buildFields?: (keyof TModel)[];
+};
+const createSpecializedBuilder = <TModel>(
+  params: TCreateSpecializedBuilderParams<TModel>
+) => {
+  const modelBuilder = Builder<TModel>({
+    type: params.type,
+    generator: params.generator,
+    transformers: createSpecializedTransformers<TModel>({
```

Comment:
Conceptually, thinking about it: we remove the need for transformers and do less heavy lifting in them. In turn we have indirection which needs to be less frequently used/visted.